test/general-spec.js

```diff
@@ -0,0 +1,131 @@
+var expect = require('expect.js');
+var simple = require('simple-mock');
+var fs = require('fs');
+var testHelper = require('../etc/test-helper.js');
+
+// Load the contents of the sync function file into a global variable called syncFunction
+eval('var syncFunction = ' + fs.readFileSync('build/sync-functions/test-general-sync-function.js').toString());
+
+// Placeholders for stubbing built-in Sync Gateway support functions.
+// More info: http://developer.couchbase.com/mobile/develop/guides/sync-gateway/sync-function-api-guide/index.html
+var requireAccess;
+var channel;
+
+describe('Functionality that is common to all documents', function() {
```

Comment:
Very minor: you may want to consider naming these descriptions and `it` specs such that, when concatenated and printed in a karma report, they generate an accurate sentence. For example, one of the test failures in this file would read:

`"Functionality that is common to all documents cannot specify whitelisted properties below the root level of the document" failed`

One option in this case is to just add a colon to your describe: `'Functionality that is common to all documents: '`